Add model and test of max31855 Cold-Junction Compensated Thermocouple-to-Digital Converter

SimonWin opened the pull request:
This pull request adds a model of the spi based max31855 Cold-Junction Compensated Thermocouple-to-Digital Converter chip (https://datasheets.maximintegrated.com/en/ds/MAX31855.pdf). It also adds a test for that model.

SimonWin commented on src/vcml/models/spi/max31855.cpp:
```diff
@@ -0,0 +1,117 @@
+    spi_in("spi_in") {
+    SC_HAS_PROCESS(max31855);
+    SC_METHOD(cs_edge);
+    sensitive << m_cs.default_event();
```
Is it intentional for gpio_target_socket that I need to call default_event here? Just doing sensitive << m_cs throws a warning and doesn't call default_event in gpio_target_socket.

janweinstock replied:
This should be fixed in the newest version

janweinstock commented:
Hi Simon!
Thanks for your PR, here are some questions before I can merge:
1. What happens when the processor sets CS and reads SPI in the same quantum, i.e., before the event has triggered?
2. Why is the chip select socket private? I can always access it through `spi_host`
3. Can we avoid having an extra process, event and gpio socket by just sampling the temperature when in `BYTE0` state?
3a. If we really need the CS (maybe the CPU resets it sometimes mid-sequence?) can we avoid the extra process by using `gpio_notify`?

SimonWin commented:
Hi,

Thanks for your questions. I think I addressed all of them in my new commit.

I made CS public and switched from using a process to using gpio_notify.
 
If you have more suggestions/questions, please let me know.
